ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -796,90 +808,154 @@ def _useRealization(self, info, rlz):
-                                   self.constraintsV)
-        self._resolveNewGeneration(traj, rlz, info)
-
-
-      # 6. Parent selection from population
-      parents = self._parentSelectionInstance(self.population,
-                                              variables=list(self.toBeSampled),
-                                              fitness=self.fitness,
-                                              kSelection=self._kSelection,
-                                              nParents=self._nParents,
-                                              rank=self.rank,
-                                              crowdDistance=self.crowdingDistance,
-                                              objVar=self._objectiveVar,
-                                              isMultiObjective = self._isMultiObjective,
-                                              )
-
-      # 7. Reproduction
-      # 7.1 Crossover
+  ## 4. Survivor selection
+      if not self._isMultiObjective:
+        survivorSelectionProcess.singleObjSurvivorSelect(self, info, rlz, traj,
+                                                         currentPop_inputs,
+                                                         currentPop_fitness,
+                                                         currentPop_objvals,
+                                                         currentPop_g)
+      else:
+        survivorSelectionProcess.multiObjSurvivorSelect(self, info, rlz, traj,
+                                                        currentPop_inputs,
+                                                        currentPop_fitness,
+                                                        currentPop_objvals,
+                                                        currentPop_g)
+      # update the age estimate for the current population for printing later
+      self.currentPop_ages = np.zeros(len(currentPop_inputs), dtype=int)
+      if self.counter > 1:
+        for i in range(len(self.currentPop_ages)):
+          indv = currentPop_inputs[i]
+          for indx, val in enumerate(self.matingPop_inputs):
+            if val.equals(indv):
+              self.currentPop_ages[i] = self.matingPop_ages[indx]
+              break
+
+  ## Single-objective post-processing
+      if not self._isMultiObjective:
+          self._collectOptPoint(rlz, currentPop_fitness, currentPop_objvals[0], currentPop_g)
+          self._resolveNewGeneration(traj, rlz, info, self.prevPop_inputs, currentPop_objvals[0], currentPop_fitness, currentPop_g)
+  ## Multi-objective post-processing
+      else:
+        # list for Rank and CD calculation
+        currentPop_fitsbysoln = datasetToDataArray(currentPop_fitness, self._objectiveVar).data.tolist()
+    ## 5. Compute the rank of current population
+        currentPop_ranks = frontUtils.rankNonDominatedFrontiers(np.array(currentPop_fitsbysoln), isFitness=True)
+        currentPop_ranks = xr.DataArray(currentPop_ranks,
```

Comment:
This is only defined if self._isMultiObjective is true. But it is used even in multiobjective and hence it will error out because it is used before assigned to a value. How did you test this? it will never run.